Multiworker QCNN tutorial

verult opened the pull request.

verult committed:
Multiworker QCNN tutorial

MichaelBroughton commented on docs/tutorials/qcnn_multiworker/inference/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+# limitations under the License.
+# ==============================================================================
+
+FROM tensorflow/tensorflow:2.3.0
+
+RUN pip install tensorflow-quantum scikit-learn google-cloud-storage
+
+COPY qcnn_inference.py ./
```
In order to do multiworker profiling, don't we need tf 2.4.1 image and tfq-nightly ?
Also don't think we need scikit-learn.... do we ? lol

MichaelBroughton reviewed (commented):
Looks great! One thing that might be worth doing: There's a lot of overlap between qccn_inference.py and qccn.py do you think it might be worth consolidating the common functions into one module and then importing that in the respective cases ?

verult commented:
Addressed comments, and restructured the directories to make a common file work. Now a single container image is shared between training and inference.

Going to merge this for now, but @MichaelBroughton still curious what you think, for future iterations
